src/lib/components/Login/BasicAuthLogin/BasicAuthLogin.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Cookie from 'js-cookie';
+
+export const isLoggedIn = function () {
+    return Cookie.get('dsauth') === "true";
+};
+
+class BasicAuthLogin extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { loginError: false };
+    }
+
+    handleLogin(password) {
+        const { appPassword } = this.props;
+        if (appPassword.includes(password)) {
+            Cookie.set('dsauth', true);
+            location.reload();
+        } else {
+            this.setState({ loginError: true });
+        }
+    }
+
+    render() {
+        if (!this.props.appPassword || !this.props.appPassword.length) {
+            return null;
+        }
+
+        return (
+            <form className="login-form" onSubmit={(e) => { e.preventDefault(); this.handleLogin(this.password.value); }}>
+                <div className="form-group">
+                    {this.state.loginError === true ? <div className="alert alert-danger">Incorrect password.</div> : null}
+                    <label htmlFor="password">Enter your Password</label>
+                    <input className="form-control" type="password" ref={(input) => { this.password = input; }} autoFocus />
+                </div>
+                <button className="btn btn-default center-block" type="submit">Log In</button>
+            </form>
+        );
+    }
+}
+BasicAuthLogin.defaultProps = {
+    appPassword: []
+
+}
+
+BasicAuthLogin.PropTypes = {
+    appPassword: PropTypes.string
+}
```

Comment:
@kelvin-muchiri the `appPassword` propTypes look like they should be an array of strings.

Reply:
> @kelvin-muchiri the `appPassword` propTypes look like they should be an array of strings.

I also thought it was string at first, till I looked at the `password` property of sites with basic auth such as service mapping and found out it's usually defined as an array